File: java/spring-boot-rabbitmq/spring-boot-rabbitmq-publisher/src/main/java/com/meridal/examples/springbootamqp/mvc/RecordingController.java
package com.meridal.examples.springbootamqp.mvc;

import com.meridal.examples.springbootamqp.service.RecordingService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/recording")
public class RecordingController {

    @Autowired
    private RecordingService service;

    @GetMapping("/{id}")
    public String getRecording(@PathVariable String id) {
        this.service.publishRecording(id);
        return "success";
    }
}

File: kotlin/kotlin-spring-boot-rabbitmq-publisher/src/main/kotlin/com/meridal/examples/kotlinspringbootamqp/mvc/RecordingController.kt
package com.meridal.examples.kotlinspringbootamqp.mvc

import com.meridal.examples.domain.Recording
import com.meridal.examples.kotlinspringbootamqp.service.RecordingService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/recording")
class RecordingController @Autowired constructor(private val service: RecordingService) {

    @PostMapping("/")
    fun saveRecording(@RequestBody recording: Recording): Recording? {
        service.publishRecording(recording)
        return recording
    }
}